### PR TITLE
python: use release branch for PyPI publishing

### PR DESCRIPTION
**Pull request by woodruffw**
### Why:

This page uses a pretty old reference for `gh-action-pypi-publish` -- I've updated it to use the `release/v1` symbolic reference instead, since that release branch has a stable API and is the encouraged branch for most users.

No associated issue.

### What's being changed (if available, include any code snippets, screenshots, or gifs):

Just the `building-and-testing-python` documentation!

### Check off the following:

- [ ] I have reviewed my changes in staging (look for the "Automatically generated comment" and click the links in the "Preview" column to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### CODE REVIEW

**Comment by woodruffw** [comment]
cc @webknjaz

**Comment by cmwilson21** [comment]
@woodruffw Thanks so much for opening a PR! I'll get this triaged for review :zap:

File: content/actions/automating-builds-and-tests/building-and-testing-python.md [hunk 1]
```diff
@@ -404,7 +404,7 @@ jobs:
       - name: Build package
         run: python -m build
       - name: Publish package
-        uses: pypa/gh-action-pypi-publish@27b31702a0e7fc50959f5ad993c78deac1bdfc29
+        uses: pypa/gh-action-pypi-publish@release/v1
         with:
           user: __token__
```
Comment (webknjaz) [inline content/actions/automating-builds-and-tests/building-and-testing-python.md:409]: Maybe also drop
```suggestion
```
while on it...

Comment (woodruffw) [inline content/actions/automating-builds-and-tests/building-and-testing-python.md:409]:
```suggestion
        with:
          user: __token__
```

File: content/actions/automating-builds-and-tests/building-and-testing-python.md [hunk 2]
```diff
@@ -404,9 +404,7 @@ jobs:
       - name: Build package
         run: python -m build
       - name: Publish package
-        uses: pypa/gh-action-pypi-publish@27b31702a0e7fc50959f5ad993c78deac1bdfc29
-        with:
-          user: __token__
+        uses: pypa/gh-action-pypi-publish@release/v1
           password: {% raw %}${{ secrets.PYPI_API_TOKEN }}{% endraw %}
```
Comment (woodruffw) [inline content/actions/automating-builds-and-tests/building-and-testing-python.md:408]:
```suggestion
        uses: pypa/gh-action-pypi-publish@release/v1
        with:
          password: {% raw %}${{ secrets.PYPI_API_TOKEN }}{% endraw %}
```

**Review by jc-clark — APPROVED** [review]
Thank you for helping us keep this updated @woodruffw! 

I also noticed the starter workflow uses the outdated version too: https://github.com/actions/starter-workflows/blob/main/ci/python-publish.yml

I'm thinking we should update this one as well. Are you interested in making a PR for that fix? 

I'll go ahead and approve this and merge it ✨

**Comment by woodruffw** [comment]
> I'm thinking we should update this one as well. Are you interested in making a PR for that fix?

Yeah, I can do that.

At this point however, both should probably be updated to reflect PyPI's support for trusted publishing: https://blog.pypi.org/posts/2023-04-20-introducing-trusted-publishers/

Those changes were originally being tracked in https://github.com/github/docs/pull/24772, but they seem to have gone somewhere internal now.

**Pull request merged**